test/Graphics/Image.test.cpp

```diff
@@ -265,6 +265,20 @@ TEST_CASE("[Graphics] sf::Image")
             CHECK(!image.loadFromFile("."));
             CHECK(!image.loadFromFile("this/does/not/exist.jpg"));
 
+            std::array<char32_t, 10> utf32name = {'X', 'f', 'a', 'k', 'e', '.', 'p', 'n', 'g', '\0'};
```

Comment:
```suggestion
            utf32name = U"Xfake.png";
```
No need for arrays. We can use string literals.
https://en.cppreference.com/w/cpp/language/string_literal

Reply:
`wchar_t` is commonly 2 bytes on Windows-like systems, and 4 bytes on Linux.
`std::filesystem::path` should accept the 4-bytes type, but will do a conversion on Windows.

Reply:
Yes, I know `wchar_t` is UTF-16 on Windows (some call it WTF-16, since it allows unpaired surrogates, some think it's broken or limited to UCS-2 but it's not).

The `char32_t` is supposed to be UTF-32 in C++ and `std::filesystem::path` is supposed to them take care of converting that to native format (16 on Windows, 8 on Linux) and stuff like `std::ifstream` is supposed to do the right thing.

I used 32 bit on purpose here, since I later put an emoji (which wouldn't fit in Windows `wchar`) in a name for a test, to make sure such characters (emojis, super niche languages, anything out of Basic Multilingual Plane) do work too, since they take 2 `wchar_t` to encode on Windows in UTF-16.

Some software used to break once emojis started becoming popular since they assumed Unicode is a 16-bit mapping, each char is single `wchar_t` on Windows (and no one ever used the higher numbered characters since they are dead language or very niche CJK stuff or such, emojis are first time these high codes are really spotted in the wild and among normal people), etc. and this test would catch such mistake. We never substr, take length with `wcslen`, etc. of the native `wchar` path on Windows, so we have 0 potential for such bugs, but it's for completness, it's a Unicode filename test, so a non-BMP character names should be tested too.

It also makes test 100% portable for any OS, it's always UTF-32 from us -> path native -> open file. We don't care or see what native format is, once UTF-32 goes inside std:: classes it's done from our point of view, it's then up to them.

Reply:
The `U` prefix connotes an UTF-32 string literal. Am I mistaken?

Reply:
It does mean `char32_t` which is UTF-32 yes and available from C++11, on my own CI I have just new tests, no fix yet, to check that the `\u` and `\U` fail as expected because they work to bring in those unicode chars. https://github.com/FRex/SFML/actions/runs/13043068592/job/36388867375

The \u is 4 digit Unicode escape and \U is an 8 digit one. Seems to be all fine.